src/util-logopenfile.c

```diff
@@ -538,19 +766,62 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
 }
 
 #ifdef HAVE_LIBHIREDIS
-static int  LogFileWriteRedis(LogFileCtx *file_ctx, const char *string, size_t string_len)
+
+/**
+ * \brief LogFileWriteRedis() writes log data to redis output.
+ * \param log_ctx Log file context allocated by caller
+ * \param string buffer with data to write
+ * \param string_len data length
+ * \retval 0 on sucess;
+ * \retval -1 on failure;
+ */
+static int LogFileWriteRedis(LogFileCtx *file_ctx, const char *string, size_t string_len)
 {
     if (file_ctx->redis == NULL) {
         SCConfLogReopenRedis(file_ctx);
         if (file_ctx->redis == NULL) {
             return -1;
         } else {
-            SCLogInfo("Reconnected to redis server");
+            SCLogInfo("Reconnected to redis server.");
         }
     }
-    /* TODO go async here ? */
+
+	SCLogRedisContext * ctx = file_ctx->redis;
+
+#if HAVE_LIBEVENT == 1
+	/* async mode on */
+    if (file_ctx->redis_setup.async) {
+
+        redisAsyncContext * redis_async = ctx->async;
+
+        if (redis_async) {
+            SCLogDebug("redis async command: %s", file_ctx->redis_setup.command);
+            redisAsyncCommand(redis_async,
+                    SCRedisAsyncCommandCallback,
+                    file_ctx,
+                    "%s %s %s",
+                    file_ctx->redis_setup.command,
+                    file_ctx->redis_setup.key,
+                    string);
+
+            redisAsyncHandleWrite(redis_async);
+            redisAsyncHandleRead(redis_async);
```

Comment:
So we will check for write answer after write. Should not it be also done when closing the connection to redis ?